script.js

```diff
@@ -0,0 +1,95 @@
+const buttons = document.querySelectorAll("button");
+const display = document.querySelector(".display");
+
+let firstNumber = "";
+let operator = "";
+let secondNumber = "";
+let result = "";
+
+const operate = (operator, a, b) => {
+  switch (operator) {
+    case "+":
+      return a + b;
+    case "-":
+      return a - b;
+    case "*":
+      return a * b;
+    case "/":
+      if (b === 0) {
+        return "Cannot divide by zero!";
+      } else {
+        return a / b;
+      }
+  }
+};
+
+const clearDisplay = () => {
+  display.textContent = "0";
+  firstNumber = "";
+  operator = "";
+  secondNumber = "";
+  result = "";
+};
+
+const updateDisplay = (value) => {
+  if (display.textContent === "0" || result !== "") {
+    display.textContent = value;
+    result = "";
+  } else {
+    display.textContent += value;
+  }
+};
+
+const backspace = () => {
+  if (secondNumber !== "") {
```

Comment:
can we avoid if-else ladder here or other places let's give a thought on this, what happen when if-else ladder is it decrease the readability of code this becomes hard to go into each nested node if condition and meshed up with logic, this should be only our last option to opt